packages/medusa/src/api-v2/admin/products/middlewares.ts

```diff
@@ -64,6 +66,19 @@ export const adminProductRoutesMiddlewares: MiddlewareRoute[] = [
       ),
     ],
   },
+  {
+    method: ["POST"],
+    matcher: "/admin/products/op/batch",
```

Comment:
Currently there is an issue with the routing, where both `/:id` and `/batch` middlewares get executed, but the route itself isn't. I think how middlewares and routes are loaded will need a revamp (my guess is that we first register all middlewares, and then all routes, but we should it route by route instead)